Adding zmmStateSupport and  AVX512F, AVX512CD, AVX512BW, AVX512DQ and AVX512VL ISAs.

DeepakRajendrakumaran opened the pull request:
This change adds logic for detecting avx512 support via `zmmStateSupport ` and the logic in `EEJitManager::SetCpuInfo()`

For specifics of `zmmStateSupport`- see section `15.2 DETECTION OF AVX-512 FOUNDATION INSTRUCTIONS` in [Intel® 64 and IA-32 Architectures Software Developer’s Manual Combined Volumes: 1, 2A, 2B, 2C, 2D, 3A, 3B, 3C, 3D, and 4](https://cdrdv2.intel.com/v1/dl/getContent/671200) for details. `14.3 DETECTION OF AVX INSTRUCTIONS` details logic used for AVX and SSE.

The other changes are to add new AVX512 ISAs. Most of these changes are autogenerated code via `src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt`

DeepakRajendrakumaran committed:
Adding avx512StateSupport().

This adds the helper functions to check if avx512 functionality is supported.

DeepakRajendrakumaran commented:
@BruceForstall @davidwrighton

MichalPetryka commented on src/coreclr/inc/clrconfigvalues.h:
```diff
@@ -765,6 +765,11 @@ RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSE41,        W("EnableSSE41"),        1
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSE42,        W("EnableSSE42"),        1, "Allows SSE4.2+ hardware intrinsics to be disabled")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSSE3,        W("EnableSSSE3"),        1, "Allows SSSE3+ hardware intrinsics to be disabled")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableX86Serialize, W("EnableX86Serialize"), 1, "Allows X86Serialize+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512F,      W("EnableAVX512F"),      1, "Allows AVX512F+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512CD,      W("EnableAVX512CD"),      1, "Allows AVX512CD+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512BW,      W("EnableAVX512BW"),      1, "Allows AVX512BW+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512DQ,      W("EnableAVX512DQ"),      1, "Allows AVX512DQ+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512VL,      W("EnableAVX512VL"),      1, "Allows AVX512VL+ hardware intrinsics to be disabled")
```
```suggestion
RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512F,      W("EnableAVX512F"),      1, "Allows AVX512F+ hardware intrinsics to be disabled")
RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512CD,     W("EnableAVX512CD"),     1, "Allows AVX512CD+ hardware intrinsics to be disabled")
RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512BW,     W("EnableAVX512BW"),     1, "Allows AVX512BW+ hardware intrinsics to be disabled")
RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512DQ,     W("EnableAVX512DQ"),     1, "Allows AVX512DQ+ hardware intrinsics to be disabled")
RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512VL,     W("EnableAVX512VL"),     1, "Allows AVX512VL+ hardware intrinsics to be disabled")
```

teo-tsirpanis commented on src/coreclr/vm/amd64/AsmHelpers.asm:
```diff
@@ -649,6 +649,22 @@ LEAF_ENTRY xmmYmmStateSupport, _TEXT
+LEAF_ENTRY zmmStateSupport, _TEXT
+        mov     ecx, 0                  ; Specify xcr0
+        xgetbv                          ; result in EDX:EAX
+        and eax, 70H
+        cmp eax, 70H                    ; check OS has enabled ZMM state support
+        jne     not_supported
+        mov     eax, 1
+        jmp     done
+    not_supported:
+        mov     eax, 0
+    done:
+        ret
```
```suggestion
        test eax, 70H                    ; check OS has enabled ZMM state support
        sete al
        movzx eax, al
        ret
```

teo-tsirpanis commented on src/coreclr/nativeaot/Runtime/amd64/GC.asm:
```diff
@@ -19,3 +19,20 @@ LEAF_ENTRY xmmYmmStateSupport, _TEXT
+LEAF_ENTRY zmmStateSupport, _TEXT
+        mov     ecx, 0                  ; Specify xcr0
+        xgetbv                          ; result in EDX:EAX
+        and eax, 70H
+        cmp eax, 70H                    ; check OS has enabled ZMM state support
+        jne     not_supported
+        mov     eax, 1
+        jmp     done
+    not_supported:
+        mov     eax, 0
+    done:
+        ret
```
```suggestion
        test eax, 70H                    ; check OS has enabled ZMM state support
        sete al
        movzx eax, al
        ret
```

teo-tsirpanis commented on src/coreclr/vm/i386/cgenx86.cpp:
```diff
@@ -1106,6 +1106,31 @@ extern "C" DWORD __stdcall xmmYmmStateSupport()
+    {
+        mov     ecx, 0                  ; Specify xcr0
+        xgetbv                          ; result in EDX:EAX
+        and eax, 70H
+        cmp eax, 70H                    ; check OS has enabled ZMM state support
+        jne     not_supported
+        mov     eax, 1
+        jmp     done
+    not_supported:
+        mov     eax, 0
+    done:
```
```suggestion
        test eax, 70H                    ; check OS has enabled ZMM state support
        sete al
        movzx eax, al
```

tannergooding commented on src/coreclr/inc/clrconfigvalues.h:
```diff
@@ -765,6 +765,11 @@ RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSE41,        W("EnableSSE41"),        1
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512CD,      W("EnableAVX512CD"),      1, "Allows AVX512CD+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512BW,      W("EnableAVX512BW"),      1, "Allows AVX512BW+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512DQ,      W("EnableAVX512DQ"),      1, "Allows AVX512DQ+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512VL,      W("EnableAVX512VL"),      1, "Allows AVX512VL+ hardware intrinsics to be disabled")
```
Due to the way `InstructionSet_*` checks work today, I think AVX512-VL might need "special handling".

In particular, each intrinsic maps to a given `InstructionSet_*` and that `InstructionSet_*` gets a bit in the crossgen metadata indicating whether a method requires or opportunistically uses a given ISA.

Since AVX512-VL is "special" in that it is its own ISA extending AVX512F, but also any other supported `AVX512-*` ISA, we need to either:
A. Explicitly declare those other ISAs similarly to what we do for X64. That is, `InstructionSet_AVX512CD_VL`, `InstructionSet_AVX512BW_VL`, etc
B. Come up with a special way to represent and check for them (such as tracking the intrinsic as `InstructionSet_AVX512CD` and having a `HW_Flag_AVX512VL` flag)

The former is consistent with how we already handle things for `X64` and somewhat simplifies the way intrinsics are checked for. It also more closely matches how the ISAs will have to be exposed from managed.

However, the latter takes up less overall "space" to encode the information around and better matches how hardware is required to check for support. Theoretically if we went with this approach we could also simplify how `InstructionSet_*_X64` was encoded/worked.

It's likely worth getting input from @davidwrighton on if one approach or the other is better.

DeepakRajendrakumaran replied:
I've made some tentative changes here while we wait for @davidwrighton . I chose option1 from above for now. Let me know if this is what you had in mind

tannergooding replied:
Thanks!

We'll want to consider the implications of each approach in relation to the `DOTNET_Enable*` environment variables and with how it impacts the ISAs users can pass into NAOT.

Hopefully we can come up with something that works well for all the scenarios here.

DeepakRajendrakumaran replied:
Also, do you have a preference for(i.e., if the 2nd looks cleaner)

```
if ((cpuidInfo[EBX] & (1 << 16)) != 0)                     // AVX512F
{
     CPUCompileFlags.Set(InstructionSet_AVX512F);

      if ((cpuidInfo[EBX] & (1 << 31)) != 0)                 // AVX512VL
      {
               CPUCompileFlags.Set(InstructionSet_AVX512F_VL);

                if ((cpuidInfo[EBX] & (1 << 30)) != 0)             // AVX512BW
                {
                        CPUCompileFlags.Set(InstructionSet_AVX512BW_VL);
                 }

                 if ((cpuidInfo[EBX] & (1 << 28)) != 0)             // AVX512CD
                 {
                         CPUCompileFlags.Set(InstructionSet_AVX512CD_VL);
                 }

                 ..............................

      }
       if ((cpuidInfo[EBX] & (1 << 30)) != 0)                 // AVX512BW
     {
            CPUCompileFlags.Set(InstructionSet_AVX512BW);
     }

     if ((cpuidInfo[EBX] & (1 << 28)) != 0)                 // AVX512CD
     {
             CPUCompileFlags.Set(InstructionSet_AVX512CD);
     }

      .............
}
```

vs


```
if ((cpuidInfo[EBX] & (1 << 16)) != 0)                     // AVX512F
 {
           CPUCompileFlags.Set(InstructionSet_AVX512F);
            bool isAVX512_VLSupported = false;

            if ((cpuidInfo[EBX] & (1 << 31)) != 0)                 // AVX512VL
            {
                   CPUCompileFlags.Set(InstructionSet_AVX512F_VL);
                   isAVX512_VLSupported = true;
             }

             if ((cpuidInfo[EBX] & (1 << 30)) != 0)                 // AVX512BW
             {
                   CPUCompileFlags.Set(InstructionSet_AVX512BW);
                    if (isAVX512_VLSupported)
                     {
                            CPUCompileFlags.Set(InstructionSet_AVX512BW_VL);// AVX512BW_VL
                      }
             }

             if ((cpuidInfo[EBX] & (1 << 28)) != 0)                 // AVX512CD
             {
                     CPUCompileFlags.Set(InstructionSet_AVX512CD);
                     if (isAVX512_VLSupported)
                     {
                             CPUCompileFlags.Set(InstructionSet_AVX512CD_VL);// AVX512CD_VL
                     }
             }
             ...........................
}
```

tannergooding replied:
I don't have a preference here. I'd guess the second is slightly less error prone since it isn't dependent on rechecking specific bits.

DeepakRajendrakumaran replied:
@tannergooding Are we still waiting to hear back on this or do I make other changes assuming AVX512*_VL design?

tannergooding replied:
I think we can assume the `AVX512*_VL` design for now and can fix up later if we need to.

davidwrighton replied:
Yes, I think the current approach will work, and if we need to change course in the future, we can address the problem then.

tannergooding commented on src/coreclr/nativeaot/Runtime/amd64/GC.asm:
```diff
@@ -19,3 +19,20 @@ LEAF_ENTRY xmmYmmStateSupport, _TEXT
         end
+
+;; extern "C" DWORD __stdcall zmmStateSupport();
+LEAF_ENTRY zmmStateSupport, _TEXT
```
Should this be named `xmmYmmZmmStateSupport` since its checking all 3-bits?

tannergooding replied:
Maybe, alternatively, `avx512StateSupport` is a better name since this this is actually checking `Opmask state`, `ZMM_Hi256 state`, and `Hi16_ZMM state`.

DeepakRajendrakumaran replied:
Changed to `avx512StateSupport`

tannergooding commented on src/coreclr/inc/clrconfigvalues.h:
```diff
@@ -765,6 +765,11 @@ RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSE41,        W("EnableSSE41"),        1
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSSE3,        W("EnableSSSE3"),        1, "Allows SSSE3+ hardware intrinsics to be disabled")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableX86Serialize, W("EnableX86Serialize"), 1, "Allows X86Serialize+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512F,      W("EnableAVX512F"),      1, "Allows AVX512F+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512CD,      W("EnableAVX512CD"),      1, "Allows AVX512CD+ hardware intrinsics to be disabled")
```
nit: These ones are all "misaligned" and have an extra whitespace in the table.

DeepakRajendrakumaran replied:
Fixed

tannergooding commented on src/coreclr/inc/clrconfigvalues.h:
```diff
@@ -765,6 +765,11 @@ RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSE41,        W("EnableSSE41"),        1
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSE42,        W("EnableSSE42"),        1, "Allows SSE4.2+ hardware intrinsics to be disabled")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSSE3,        W("EnableSSSE3"),        1, "Allows SSSE3+ hardware intrinsics to be disabled")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableX86Serialize, W("EnableX86Serialize"), 1, "Allows X86Serialize+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512F,      W("EnableAVX512F"),      1, "Allows AVX512F+ hardware intrinsics to be disabled")
```
There's a secondary set of config values used internally by the JIT for `AltJit` scenarios in `src/coreclr/jit/jitconfigvalues.h`. These should be kept in sync with the values declared here.

DeepakRajendrakumaran replied:
Added to `src/coreclr/jit/jitconfigvalues.h`

tannergooding commented on src/coreclr/nativeaot/Runtime/amd64/GC.asm:
```diff
@@ -19,3 +19,20 @@ LEAF_ENTRY xmmYmmStateSupport, _TEXT
+        mov     ecx, 0                  ; Specify xcr0
+        xgetbv                          ; result in EDX:EAX
+        and eax, 70H
+        cmp eax, 70H                    ; check OS has enabled ZMM state support
```
I think this needs to be `E0H`. The bits we want to check are `XCR0[7:5]` where:
* XCR0[0] is associated with x87 state
* XCR0[1] is associated with SSE state
* XCR0[2] is associated with AVX state
* XCR0[4:3] are associated with MPX state
* XCR0[7:5] are associated with AVX-512 state
* XCR0[8] is reserved
* XCR0[9] is associated with PKRU state
* XCR0[63:10] isreserved.

Additionally, given that we technically need to check both `XCR0[7:5] == 111b` and `XCR0[2:1] == 11b`, it might be better to just check against `E6H` to ensure that any call to this checks the necessary bits and that we don't strictly require calling both `xmmYmmStateSupport` and `zmmStateSupport` independently.

DeepakRajendrakumaran replied:
That is a bad mistake on my part. Thanks for catching it and pointing it out. I went with `0E6H` so what we check for all of `SSE AVX and AVX512` states. This reflected in the change in `EEJitManager::SetCpuInfo()`

tannergooding commented on src/coreclr/vm/codeman.cpp:
```diff
@@ -1467,6 +1467,36 @@ void EEJitManager::SetCpuInfo()
                                         }
                                     }
+
+                                    if (zmmStateSupport() == 1)                                             // XGETBV XRC0[7:5] == 111
```
This should probably be nested after the `CPUCompileFlags.Set(InstructionSet_AVX2);` for consistency with how the other checks are done and to short circuit the logic if AVX2 isn't supported.

DeepakRajendrakumaran replied:
Changed. The `AVX512VL` logic in there is my interpretation of what you meant in terms of `AVX512*_VL` flag usage. Let me know if I misunderstood.

DeepakRajendrakumaran commented on src/coreclr/inc/clrconfigvalues.h:
```diff
@@ -765,6 +765,11 @@ RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSE41,        W("EnableSSE41"),        1
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSE42,        W("EnableSSE42"),        1, "Allows SSE4.2+ hardware intrinsics to be disabled")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSSE3,        W("EnableSSSE3"),        1, "Allows SSSE3+ hardware intrinsics to be disabled")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableX86Serialize, W("EnableX86Serialize"), 1, "Allows X86Serialize+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512F,      W("EnableAVX512F"),      1, "Allows AVX512F+ hardware intrinsics to be disabled")
```
RE-ordered alphabetically. Will add additional ones after the `AVX512VL `discussion is resolved

DeepakRajendrakumaran commented on src/coreclr/vm/codeman.cpp:
```diff
@@ -1467,6 +1467,36 @@ void EEJitManager::SetCpuInfo()
                                         }
                                     }
+
+                                    if (zmmStateSupport() == 1)                                             // XGETBV XRC0[7:5] == 111
```
Added the `DoesOSSupportAVX512()` as suggested

tannergooding commented on src/coreclr/inc/clrconfigvalues.h:
```diff
@@ -749,6 +749,14 @@ RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableHWIntrinsic,  W("EnableHWIntrinsic"),  1
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX,          W("EnableAVX"),          1, "Allows AVX+ hardware intrinsics to be disabled")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX2,         W("EnableAVX2"),         1, "Allows AVX2+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512BW,     W("EnableAVX512BW"),     1, "Allows AVX512BW+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512BW_VL,  W("EnableAVX512BW_VL"),  1, "Allows AVX512BW+ and AVX512VL+ hardware intrinsics to be disabled")
```
I wonder if we can reword the "description" better. The concern is that
> "Allows AVX512BW+ and AVX512VL+ hardware intrinsics to be disabled"

Is interpreted as equivalent to `EnableAVX512BW=0` + `EnableAVX512VL=0`, when in practice its independent.

Maybe just this would work, for consistency?
> Allows AVX512BW_VL+ hardware intrinsics to be disabled

tannergooding replied:
This comment is still pending.

DeepakRajendrakumaran replied:
Done

DeepakRajendrakumaran commented:
@tannergooding I cannot figure out where/how `XmmYmmStateSupport ` in `src/coreclr/pal/src/arch/amd64/processor.cpp` is used. And why it's `XmmYmmStateSupport ` as opposed to `xmmYmmStateSupport `. Are you familiar with this call?

tannergooding commented on src/coreclr/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt:
```diff
@@ -91,6 +107,15 @@ implication        ,X86   ,AVXVNNI   ,AVX2
+implication        ,X86   ,AVX512BW_VL,  AVX512F_VL
+implication        ,X86   ,AVX512DQ,  AVX512F
+implication        ,X86   ,AVX512DQ_VL,  AVX512F_VL
+
```
Near the bottom of the file, we should also add the following (expanding `avx512vl` as appropriate) to keep parity with clang:
```txt
instructionsetgroup ,x86-x64-v4 ,X64 X86 ,x86-x64-v3 avx512f avx512bw avx512cd avx512vl
```

tannergooding replied:
This comment is still pending.

DeepakRajendrakumaran replied:
Done

DeepakRajendrakumaran replied:
Done

tannergooding commented:
> I cannot figure out where/how XmmYmmStateSupport in src/coreclr/pal/src/arch/amd64/processor.cpp is used. And why it's XmmYmmStateSupport as opposed to xmmYmmStateSupport . Are you familiar with this call?

I don't see any existing consumers, it might be dead code.

tannergooding commented:
There's still a couple of pending comments above and we need to revise the JIT/EE Version Guid here: https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/jiteeversionguid.h

Otherwise the changes look good/correct to me and I think we can adjust the `_VL` handling if needed later. Would still be good if someone from @dotnet/crossgen-contrib could review those changes as well, however.

DeepakRajendrakumaran commented on src/coreclr/vm/codeman.cpp:
```diff
@@ -1467,6 +1467,36 @@ void EEJitManager::SetCpuInfo()
+                                    {
+                                        __cpuidex(cpuidInfo, 0x00000007, 0x00000000);
+
+                                        if ((cpuidInfo[EBX] & (1 << 16)) != 0)                              // AVX512F
```
Have added comments.

DeepakRajendrakumaran commented:
> There's still a couple of pending comments above and we need to revise the JIT/EE Version Guid here: https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/jiteeversionguid.h
> 
> Otherwise the changes look good/correct to me and I think we can adjust the `_VL` handling if needed later. Would still be good if someone from @dotnet/crossgen-contrib could review those changes as well, however.

Have added the requested changes.

BruceForstall commented:
Tagging the group... just to see it work... @dotnet/avx512-contrib

DeepakRajendrakumaran commented:
> Tagging the group... just to see it work... @dotnet/avx512-contrib

Works for me. Got an email notification

DeepakRajendrakumaran commented:
> There's still a couple of pending comments above and we need to revise the JIT/EE Version Guid here: https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/jiteeversionguid.h
> 
> Otherwise the changes look good/correct to me and I think we can adjust the `_VL` handling if needed later. Would still be good if someone from @dotnet/crossgen-contrib could review those changes as well, however.

@tannergooding I had originally missed your comment re JIT/EE Version Guid. I have updated it now following instruction in the file(ran "uuidgen.exe -s" locally and replaced the GUID in file with that). Please let me know if there is anything else needed from us to push this forward.

tannergooding commented on src/coreclr/tools/Common/JitInterface/CorInfoBase.cs:
```diff
@@ -2776,3 +2777,4 @@ private static IntPtr GetUnmanagedCallbacks()
         }
     }
 }
+
```
Looks like a lot of failures caused by this newline.

tannergooding replied:
Seems to be an enforced code style analyzer, possibly specific to the `JitInterface` cs files, since it isn't one we enforce in corelib or more broadly to my knowledge.

DeepakRajendrakumaran replied:
I have fixed that. I do still see some superpmi failures. I do not expect any diffs in code since we are not really modifying codegen. Should I look into these or are these known issues? This looks like a infra failure from what I understand but do let me know if I have misunderstood the below logs(Esp following - **_No Azure Storage MCH files to download from eb8352bd-0a13-4b5b-badb-58f9ecc40c44/linux/arm64/_**)


```
Generating script.
2022-09-13T23:15:27.2899347Z ========================== Starting Command Output ===========================
2022-09-13T23:15:27.3168856Z ##[command]"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_work\_temp\35ae31c7-4cfd-466d-bdda-8e97e6756db0.ps1'"
2022-09-13T23:15:30.6143525Z Setting NUGET enhanced retry environment variables
2022-09-13T23:15:30.6657992Z Setting NUGET enhanced retry environment variables
2022-09-13T23:15:31.0448738Z D:\a\_work\1\s\.dotnet\sdk\7.0.100-preview.7.22377.5\MSBuild.dll /nologo -maxcpucount /m /maxcpucount -verbosity:m /v:minimal /bl:D:\a\_work\1\s/artifacts/log/SendToHelix.binlog /clp:Summary /logger:D:\a\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\8.0.0-beta.22455.1\tools\net7.0\Microsoft.DotNet.ArcadeLogging.dll /nr:False /p:ContinuousIntegrationBuild=True /p:TreatWarningsAsErrors=false /p:TargetArchitecture=x64 /p:TargetOS=windows /p:TargetOSSubgroup= /p:Configuration=checked D:\a\_work\1\s/src/coreclr/scripts/superpmi-replay.proj
2022-09-13T23:15:34.2433810Z   Sending Job to Windows.10.Amd64.Open...
2022-09-13T23:15:36.6819223Z   Sent Helix Job; see work items at https://helix.dot.net/api/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems?api-version=2019-06-17
2022-09-13T23:15:37.7129910Z   Waiting for completion of job 6935e4d0-ad31-4de8-843e-519be36c722a on Windows.10.Amd64.Open
2022-09-13T23:37:01.6476418Z   Job 6935e4d0-ad31-4de8-843e-519be36c722a on Windows.10.Amd64.Open is completed with 11 finished work items.
2022-09-13T23:37:07.7705162Z   Downloading result files for job 6935e4d0-ad31-4de8-843e-519be36c722a
2022-09-13T23:37:07.8817262Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item linux-arm64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.8820413Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/linux-arm64-1/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.8947072Z ##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item linux-arm64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed.
Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/linux-arm64-1/console
2022-09-13T23:37:07.8967391Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item linux-arm64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.8972720Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/linux-arm64-2/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.8978502Z ##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item linux-arm64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed.
Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/linux-arm64-2/console
2022-09-13T23:37:07.8982267Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item osx-arm64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.8984930Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/osx-arm64-1/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.8995906Z ##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item osx-arm64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed.
Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/osx-arm64-1/console
2022-09-13T23:37:07.8998310Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item osx-arm64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9000977Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/osx-arm64-2/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9010591Z ##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item osx-arm64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed.
Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/osx-arm64-2/console
2022-09-13T23:37:07.9014864Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item unix-x64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9017581Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/unix-x64-1/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9022925Z ##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item unix-x64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed.
Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/unix-x64-1/console
2022-09-13T23:37:07.9025184Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item unix-x64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9027264Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/unix-x64-2/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9032253Z ##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item unix-x64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed.
Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/unix-x64-2/console
2022-09-13T23:37:07.9034482Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item win-arm64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9035897Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-arm64-1/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9040554Z ##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item win-arm64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed.
Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-arm64-1/console
2022-09-13T23:37:07.9042639Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item win-arm64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9043991Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-arm64-2/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9050019Z ##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item win-arm64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed.
Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-arm64-2/console
2022-09-13T23:37:07.9052673Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item win-x64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9054225Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-x64-1/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9059145Z ##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item win-x64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed.
Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-x64-1/console
2022-09-13T23:37:07.9061579Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item win-x64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9063244Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-x64-2/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9067693Z ##[error].packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Work item win-x64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed.
Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-x64-2/console
2022-09-13T23:37:07.9069142Z 
2022-09-13T23:37:07.9069434Z Build FAILED.
2022-09-13T23:37:07.9069722Z 
2022-09-13T23:37:07.9070536Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item linux-arm64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9071891Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/linux-arm64-1/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9073394Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item linux-arm64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9074761Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/linux-arm64-2/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9076273Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item osx-arm64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9077833Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/osx-arm64-1/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9079305Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item osx-arm64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9080653Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/osx-arm64-2/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9081985Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item unix-x64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9083316Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/unix-x64-1/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9084730Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item unix-x64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9086095Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/unix-x64-2/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9087428Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item win-arm64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9088913Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-arm64-1/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9090461Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item win-arm64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9091834Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-arm64-2/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9093351Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item win-x64-1 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9094878Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-x64-1/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9096338Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Work item win-x64-2 in job 6935e4d0-ad31-4de8-843e-519be36c722a has failed. [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9097798Z D:\a\_work\1\s\.packages\microsoft.dotnet.helix.sdk\8.0.0-beta.22455.1\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(90,5): error : Failure log: https://helix.dot.net/api/2019-06-17/jobs/6935e4d0-ad31-4de8-843e-519be36c722a/workitems/win-x64-2/console [D:\a\_work\1\s\src\coreclr\scripts\superpmi-replay.proj]
2022-09-13T23:37:07.9098671Z     0 Warning(s)
2022-09-13T23:37:07.9098971Z     10 Error(s)
2022-09-13T23:37:07.9099085Z 
2022-09-13T23:37:07.9099419Z Time Elapsed 00:21:36.76
2022-09-13T23:37:07.9660519Z Build failed with exit code 1. Check errors above.
```


Looking at the logs mentioned there


```
Console log: 'linux-arm64-1' from job 6935e4d0-ad31-4de8-843e-519be36c722a workitem 9b195ed3-9043-4207-9503-c4d5e50d3e2d (windows.10.amd64.open.rt) executed on machine a0026IJ running Windows-10-10.0.14393-SP0

C:\h\w\AA030949\w\9F1D08E5\e>C:\python3\python.exe C:\h\w\AA030949\p\superpmi_replay.py -jit_directory C:\h\w\AA030949\p -arch arm64 -platform Linux -partition 1 -partition_count 2 -log_directory C:\h\w\AA030949\w\9F1D08E5\uploads 
Running superpmi.py download
Running: C:\python3\python.exe C:\h\w\AA030949\p\superpmi.py download --no_progress -target_os Linux -target_arch arm64 -core_root C:\h\w\AA030949\p -spmi_location C:\h\w\AA030949\p\artifacts\spmi -log_level debug
[23:36:07] ================ Logging to C:\h\w\AA030949\p\artifacts\spmi\superpmi.log

[23:36:07] Using mcs.exe from Core_Root: C:\h\w\AA030949\p\mcs.exe

[23:36:07] Using JIT/EE Version from mcs: eb8352bd-0a13-4b5b-badb-58f9ecc40c44

[23:36:07] SuperPMI download

[23:36:07] ------------------------------------------------------------

[23:36:07] Start time: 23:36:07

**_No Azure Storage MCH files to download from eb8352bd-0a13-4b5b-badb-58f9ecc40c44/linux/arm64/_**

[23:36:08] Finish time: 23:36:08

[23:36:08] Elapsed time: 0:00:00.658097

```

tannergooding replied:
> Correction - looks like non exists for this version coz GUID got updated

Right. SuperPMI isn't robust against version GUID changes and so won't give any valid results for this PR.

DeepakRajendrakumaran replied:
Are you comfortable with leaving it as is or do I run superpmi locally without the GUID change to verify?

davidwrighton commented:
I've approved, but I'd like to see an approval from @tannergooding as well before we merge.

tannergooding commented on src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs:
```diff
@@ -801,6 +977,8 @@ private static Dictionary<(string, TargetArchitecture), string> AllInstructionSe
                 { ("skylake",    TargetArchitecture.X64),   "x86-x64-v3" },
                 { ("skylake",    TargetArchitecture.X86),   "x86-x64-v3" },
+                { ("x86-x64-v4", TargetArchitecture.X64),   "x86-x64-v3 avx512f avx512bw avx512cd avx512vl" },
+                { ("x86-x64-v4", TargetArchitecture.X86),   "x86-x64-v3 avx512f avx512bw avx512cd avx512vl" },
```
Should `avx512vl` be `avx512f_vl`?

DeepakRajendrakumaran replied:
So that would mean multiple new instruction groups, yes?

tannergooding replied:
No, I think we just need to modify `x64-x64-v4` to include the full set of ISAs it implies are supported.

DeepakRajendrakumaran replied:
Done.

tannergooding reviewed (approved):
LGTM, minus the question about the NAOT instruction set names

jkotas commented on src/coreclr/nativeaot/Runtime/amd64/GC.asm:
```diff
@@ -19,3 +19,20 @@ LEAF_ENTRY xmmYmmStateSupport, _TEXT
         end
+
+;; extern "C" DWORD __stdcall avx512StateSupport();
+LEAF_ENTRY avx512StateSupport, _TEXT
```
This needs to be hooked up here: https://github.com/dotnet/runtime/blob/0a60d952587a405d00c58e327208ebc55ba9e7c0/src/coreclr/nativeaot/Runtime/startup.cpp#L166

tannergooding replied:
Looks like the AVX512 ISA checks need to be mirrored there as well.

DeepakRajendrakumaran replied:
And the 'end' was in the wrong place :) . Have added the checks in startup.cpp

DeepakRajendrakumaran committed:
Adding AVX512F, AVX512CD, AVX512BW, AVX512DQ and AVX512VL ISAs.

DeepakRajendrakumaran committed:
Updating JIT/EE Version Guid.

tannergooding commented:
Failures are #75667 

super-pmi failures are due to the JIT/EE version guid update

timeouts are on a subset of arm/arm64 machines and are happening everywhere, this change is explicitly around xarch.
